woocommerce/woocommerce-gateway-stripe
# [Wallets] Display payment declines on the checkout and re-enable the Place order button after closing the modal

Fixes #3168

## Changes proposed in this Pull Request:

This PR fixes 2 issues with Wallet payment methods (Cash App and WeChat).

1. If you close the QR code modal on the Block checkout, the **Place order** button remains disabled and the customer would need to refresh the page to bring it back. 
2. If you fail the payment, you're redirected to the order received page rather than being left on the checkout.

With this PR if you fail the wallet payment it will now leave you on the checkout and display the error received from Stripe. If you close the modal, the block checkout is reinitialised so you can reattempt payment. 

## Testing instructions

1. As a merchant, enable **WeChat** and **Cash App**.
2. Create a Checkout Block page, if you don't have one already.
3. As a shopper, add a product to the cart and go to the Block Checkout page
4. Select **WeChat** or **Cash App** as the payment method
5. Click on **Place Order**
6. Close the modal without proceeding with the payment
    7. On develop the Place Order button remains disabled. It's not possible to place the order until you reload the page.
    8. On this branch the place order button should be ready to go again. 

| `develop` | This branch |
|--------|--------|
| <img width="1105" alt="Screenshot 2024-06-07 at 10 01 15 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/d7996c57-efe3-4012-a563-0710223dea42"> | <img width="1103" alt="Screenshot 2024-06-07 at 9 59 27 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/40250c0f-dcca-4e33-b03e-61d89ec0662d"> | 

1. As a shopper, add a product to the cart and go to the Block Checkout page
4. Select **WeChat** or **Cash App** as the payment method
5. Click on **Place Order**
6. Scan the QR code on your mobile device.
7. At Stripe click the **Fail test payment** button.
   - On `develop` you would be redirected to the order received page. Where you would need to  re-navigate to the checkout to reattempt payment.
   - On this branch you should receive the failed notice on the checkout. 

<p align="center">
<img width="1115" alt="Screenshot 2024-06-07 at 10 06 01 PM" src="https://github.com/woocommerce/woocommerce-gateway-stripe/assets/8490476/507a60d4-1980-4737-936e-d6c9368000e5">
<p>




---


**Post merge**
